.github/workflows/tests.yml

```diff
@@ -0,0 +1,78 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - master
+      - release
+  pull_request:
+    branches:
+      - '*'
```

Comment:
This will make it run for everyone everywhere! We probably wanna limit this one to just the cronjob.

Reply:
My thinking was that, by running in on PRs, we can catch changes in batch_processing values, then make decisions about  within that PR:

* If the change is undesired, then we can edit the PR to make sure batch_processing stays the same.
* If the change was expected, then we can update the values in the cached .csv files within the same PR.

Reply:
Okay, cool, so that was intentional.

I think batch_processing takes like 10 minutes to run though, so it might not be a good candidate for an every-day kinda thing. But then people need to know how to figure out what it means when the error kicks them after their PR has been merged.

Reply:
> I think batch_processing takes like 10 minutes to run though, so it might not be a good candidate for an every-day kinda thing.

Good point... the total time for the full workflow is ~21m, while with #3125, the slowest job takes only 10m. 

The alternative could be making batch_processing a non-required check on PRs. That way it still runs, and we can be aware of any changes, but also intentionally merge PRs without waiting for it to finish on a case-by-case basis.

Reply:
Oh of course, that's the best of both worlds. Assuming we can afford that.

On Travis, our CPU time was throttled. While working on this I did manage to hit the throttle, but it seems that Actions is a lot more generous: I only hit it after having something like 30 hung processes; on Travis, the throttle is more like... 4 per account. I would be worried about tens of 10-minute checks running in parallel eating into our throttle.

This is why I wanted to give this workflow a different name (https://github.com/neuropoly/spinalcordtoolbox/pull/3198/files#r564848578), because I wasn't imagining we could run it in the same batch; and we probably don't want to run it on all the platforms, either. Platform differences *could* cause numerical differences, but I think `batch_processing.sh` is more of a lumbering large-scale check and it's a lot more likely that changes will come from changed algorithms than from ANTS being miscompiled but only for MacOS. Of course, if that *did* happen and we missed it it would be very bad.

Maybe @jcohenadad can make a ruling on how often we should plan to run `batch_processing.sh`.